code/MSSQLSchemaManager.php

```diff
@@ -390,7 +390,7 @@ protected function alterTableAlterColumn($tableName, $colName, $colSpec)
         // First, we split the column specifications into parts
         // TODO: this returns an empty array for the following string: int(11) not null auto_increment
         //		 on second thoughts, why is an auto_increment field being passed through?
-        $pattern = '/^(?<definition>[\w()]+)\s?(?<null>(?:not\s)?null)?\s?(?<default>default\s[\w\']+)?\s?(?<check>check\s?[\w()\'",\s]+)?$/i';
+        $pattern = '/^(?<definition>[\w()]+)\s?(?<null>(?:not\s)?null)?\s?(?<default>default\s[\w\'\\\\]+)?\s?(?<check>check\s?[\w()\'"\\\\,\s]+)?/i';
```

Comment:
What other characters could the default have? Spaces, underscore, etc? Feels like regexp is totally the wrong way to go about this hah.

Maybe match word characters without quotes, but also match a quoted entity that contains everything except a quote?

Reply:
Something like this https://stackoverflow.com/questions/375104/how-can-i-match-a-quote-delimited-string-with-a-regex

Reply:
May as well unit test it right? :P